### ISSUE TITLE
Ability to configure embeddings dimension size

### ISSUE THREAD

**Issue by jamesbraza** [ollama/ollama#651]
Moving from `OllamaEmbeddings(model="llama2:13b")` to `OllamaEmbeddings(model="llama2:7b")`, I am now getting a shape mismatch in my embeddings:

```none
ValueError: shapes (4096,) and (5120,) not aligned: 4096 (dim 0) != 5120 (dim 0)
```

So the `7b` embeddings is slightly smaller (4096) than `13b` embeddings (5120).  Is there an argument or parameter I can use to control the embedding size?

I would like to artificially switch `7b` to use 5120, so I can not rebuild my vector store.

**Comment by kittycattoys** [comment]
This functionality from Nomic is nice. I am using TinyLlama / Llama2 for embeddings but would like the option to change dimensionality like Nomic has:   from nomic import embed response = embed.text(
     texts=[prompt],
     model='nomic-embed-text-v1.5',
     task_type='search_document',
     dimensionality=768
)   Is this enhancement possible?

**Comment by bilunsun** [comment]
I second this.
> This functionality from Nomic is nice. I am using TinyLlama / Llama2 for embeddings but would like the option to change dimensionality like Nomic has: from nomic import embed response = embed.text( texts=[prompt], model='nomic-embed-text-v1.5', task_type='search_document', dimensionality=768 ) Is this enhancement possible?

In their latest v1.5 [HuggingFace page](https://huggingface.co/nomic-ai/nomic-embed-text-v1.5), the user should be able to configure both the `task_type` and `dimensionality`. However, I have not been able to set these arguments with Ollama.

**Comment by kenneth-liao** [comment]
Agree, this type of functionality is critical for optimizing the embedding process and database size when storing embeddings.

**Comment by raffaeler** [comment]
FYI if the model has Matryoshka capabilities, cutting the vector to the desired size is straightforward. You just keep the most significant portion:
1. take the [0:length] part
2. normalize the result vector

**Comment by saadshaikh3** [comment]
Any update on this feature ?

**Comment by noahfraiture** [comment]
Still nothing ?

**Comment by PhilippeCalvez** [comment]
On my side the problem was due to a rate limited litellm proxy that I'm using. I've installed the model nomic-embed-text on a local ollama server. It works but I had to specify the embedding dimension.

`# Define the Crew
crew = Crew(
    agents=agents,
    tasks=tasks,
    process=Process.sequential,
    verbose=True,
    memory=True,
    embedder={
        "provider": "ollama",
        "config": {
            "model": 'nomic-embed-text',
            "api_base": "http://localhost:11434/api/embed"
        }
    },
    embedding_dimension=768
)
`

**Comment by rick-github** [comment]
https://github.com/ollama/ollama/pull/12242

**Comment by jamesbraza** [comment]
@rick-github feel free to close this issue if you think that PR resolves it. I made this request two years ago, I don't really have the code to test this any more
